sequencer/src/lib.rs

```diff
@@ -315,7 +316,7 @@ pub async fn init_node(
     let state_ver_keys = (0..num_nodes)
         .map(|i| StateKeyPair::generate_from_seed_indexed(config.seed, i as u64).ver_key())
         .collect::<Vec<_>>();
-    let state_key_pair = config.config.my_own_validator_config.state_key_pair.clone();
+    let state_key_pair = StateKeyPair::generate_from_seed_indexed(config.seed, node_index);
```

Comment:
Why this change?

Reply:
I'm building on top of #1041. 

https://github.com/EspressoSystems/espresso-sequencer/pull/1041/files#diff-d10b6700d92426a01587eb55cefd08511231a87147b0616810bded2dcc2dbb74R318

But I guess that makes it hard to review. I can move the target branch or make another PR with #1099 as the target. or some other approach...

Reply:
Ooh yeah moving the target branch so we merge into #1041 instead of accumulating the changes would make it much easier to review

Reply:
I do have this question though for @alxiong or @mrain : how come we're re-generating the key pair now instead of using the one from `my_own_validator_config`?

Reply:
I was planing to remove the `StateKeyPair` from the consensus scope. But it's not fully decided yet.

Reply:
Oh gotcha

Reply:
Oh I really don't recommend basing on #1041 for this hotshot upgrade, b/c we have so many breaking changes there. Since you have already done it, remember to revert the changes here: [f48af9a](https://github.com/EspressoSystems/espresso-sequencer/pull/1066/commits/f48af9a7bb64cf26ed113c48e130af9a8cbede68)

Reply:
:thinking:

Reply:
> I do have this question though for @alxiong or @mrain : how come we're re-generating the key pair now instead of using the one from my_own_validator_config?
> 
> [mrain](https://github.com/mrain) [Feb 9, 2024](https://github.com/EspressoSystems/espresso-sequencer/pull/1066#discussion_r1483666648)
> 
> I was planing to remove the StateKeyPair from the consensus scope. But it's not fully decided yet.

I don't understand the change as well. but judging from the latest tip of this PR, this particular change is reverted?

also @mrain wdym by "removing the StateKeyPair from the consensus scope?

Reply:
> Oh I really don't recommend basing on #1041 for this hotshot upgrade, b/c we have so many breaking changes there. Since you have already done it, remember to revert the changes here: [f48af9a](https://github.com/EspressoSystems/espresso-sequencer/pull/1066/commits/f48af9a7bb64cf26ed113c48e130af9a8cbede68)

I spent some time trying to disentangle our PRs. The problem is that we have mutual inter-dependencies. The #1041 is making updates that this PR needs in order to build. I think what we can do is integrate this PR with #1041 first, then do the merge to main. Any objection @jbearer @alxiong @mrain ?

Reply:
No objection.